src/Microsoft.Health.Dicom.SqlServer/Features/Retrieve/SqlInstanceStoreV47.cs

```diff
@@ -0,0 +1,63 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Data.SqlClient;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Models;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Client;
+
+namespace Microsoft.Health.Dicom.SqlServer.Features.Retrieve;
+
+internal class SqlInstanceStoreV47 : SqlInstanceStoreV46
+{
+    public SqlInstanceStoreV47(SqlConnectionWrapperFactory sqlConnectionWrapperFactory) : base(sqlConnectionWrapperFactory)
+    {
+    }
+
+    public override SchemaVersion Version => SchemaVersion.V47;
+
+
+    public override async Task<IReadOnlyList<WatermarkRange>> GetInstanceBatchesByTimeStampAsync(
```

Comment:
As I ask in `GetInstanceBatchesByTimestamp.sql`, can this be combined with the existing `GetInstanceBatches`?

Reply:
Yes, same as comment as before. I am thinking of fixing the customer and refactoring later for perfection, if you insist reusing I put the effort now.